webapp/src/snippetBuilder.tsx

```diff
@@ -0,0 +1,336 @@
+/// <reference path="../../built/pxtlib.d.ts" />
+
+import * as React from "react";
+import * as data from "./data";
+import * as sui from "./sui";
+import * as md from "./marked";
+import * as compiler from './compiler';
+
+type ISettingsProps = pxt.editor.ISettingsProps;
+
+interface SnippetBuilderProps extends ISettingsProps {
+    mainWorkspace: Blockly.Workspace;
+    config: pxt.SnippetConfig;
+}
+
+type AnswerTypes = any; // Should include custom answer types for number, enums, string, image
+
+interface GoToParameters {
+    [tokenType: string]: number;
+}
+
+interface GoToOptions {
+    question: number;
+    parameters?: GoToParameters;
+}
+
+interface QuestionInput {
+    answerToken: string;
+    defaultAnswer: AnswerTypes;
+    type?: string;
+    label?: string;
+}
+
+interface Questions {
+    title: string;
+    output?: string;
+    goto?: GoToOptions;
+    inputs: QuestionInput[];
+}
+
+interface SnippetConfig {
```

Comment:
Is it the same as in pxtarget.d.ts?

Reply:
Yeah I just updated them to use the pxtarget ones.